hal/halx86/minihal/CMakeLists.txt

```diff
@@ -1,30 +1,89 @@
 
-list(APPEND MINI_HAL_SOURCE
-    ../generic/portio.c
-    ../legacy/bus/bushndlr.c
-    ../legacy/bus/cmosbus.c
-    ../legacy/bus/isabus.c
-    ../legacy/bus/pcibus.c
-    ../legacy/bus/sysbus.c
-    ../legacy/bussupp.c
-    ../generic/beep.c
-    ../generic/bios.c
-    ../generic/clock.c
-    ../generic/cmos.c
-    ../generic/dma.c
-    ../generic/display.c
-    ../generic/drive.c
-    ../generic/misc.c
-    ../generic/profil.c
-    ../generic/reboot.c
-    ../generic/spinlock.c
-    ../generic/sysinfo.c
-    ../generic/timer.c
-    ../generic/usage.c
-    ../up/halinit_mini.c
-    ../up/pic.c
-    ../up/processor.c
-    ../include/hal.h)
+if(SARCH STREQUAL "xbox")
```

Comment:
These three lists have in common a lot of files; wouldn't it be possible to keep the generic `list(APPEND MINI_HAL_SOURCE` as it was before, outside the SARCH ifdefs, but with only really the common files, and then,
do these `if(SARCH STREQUAL ...` and append only the other files that are platform-specific?